payments-core/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -86,27 +89,17 @@ internal abstract class BaseAddPaymentMethodFragment(
 
         childFragmentManager.addFragmentOnAttachListener { _, fragment ->
             (fragment as? ComposeFormDataCollectionFragment)?.let { formFragment ->
-                formFragment.paramMapLiveData.observe(viewLifecycleOwner) { paramMap ->
-                    sheetViewModel.updateSelection(
-                        paramMap?.run {
-                            PaymentMethodCreateParams.Type.fromCode(paramMap["type"] as String)
-                                ?.let {
-                                    PaymentMethodCreateParams(
-                                        it,
-                                        overrideParamMap = paramMap,
-                                        productUsage = setOf("PaymentSheet")
-                                    )
-                                }
-                        }?.let {
-                            PaymentSelection.New.GenericPaymentMethod(
-                                selectedPaymentMethod.displayNameResource,
-                                selectedPaymentMethod.iconResource,
-                                it,
-                                false
-                            )
-                        }
+                formFragment.formViewModel.completeFormValues.map { formFieldValues ->
```

Comment:
Not blocking, but it feels like this logic should mostly live in `ComposeFormDataCollectionFragment` or its ViewModel to enforce separation of concerns.

Is there for a way to notify `ComposeFormDataCollectionFragment` when it is attached, and then have it emit some value?

Or if this is just constantly listening for the latest emission of `completeFormValues` you could use Fragment Result Listener API:


```
class ComposeFormDataCollectionFragment {
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        viewLifecycleOwner.lifecycleScope.launch {
            formViewModel.completeFormValues.collect {
                setFragmentResult(
                    "request_key_form_field_values",
                    bundleOf("values" to it)
                )
            }
        }
    }
}

class BaseAddPaymentMethodFragment {
  
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        childFragmentManager.setFragmentResultListener(
            ComposeFormDataCollectionFragment.REQUEST_KEY_FORM_FIELD_VALUES,
            viewLifecycleOwner
        ) { _, bundle ->
          bundle.getParcelable("values")
        }
    }
}
```

Reply:
I agree, the reason it has to live here now is because our paymentsheet code is split between the core and paymentsheet module, so something in Core needs to know about the PaymentMethodCreateParams